test/boost/chunked_vector_test.cc

```diff
@@ -413,3 +414,15 @@ BOOST_AUTO_TEST_CASE(tests_copy_constructor_exception_safety) {
         BOOST_REQUIRE_EQUAL(checker.live_objects(), 0);
     }
 }
+
+BOOST_AUTO_TEST_CASE(test_initializer_list_ctor) {
+    constexpr size_t chunk_size = 2;
+    auto ilist = std::initializer_list<data_value>{1, "two", 3.0};
+    auto vec = utils::chunked_vector<data_value, chunk_size>(ilist);
+    BOOST_REQUIRE_EQUAL(vec.size(), ilist.size());
+    auto it = ilist.begin();
+    auto vit = vec.begin();
+    for (; it != ilist.end(); ++it, ++vit) {
+        BOOST_REQUIRE(*it == *vit);
+    }
```

Comment:
nitpick: at the end of the loop you can also BOOST_REQUIRE that it == vec.end(). Probably not important though (since you already checked the size).

Reply:
right.  it can replace the size requirement, or just added on top of it